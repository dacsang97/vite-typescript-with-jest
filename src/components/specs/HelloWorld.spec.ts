import { mount } from '@vue/test-utils'
import HelloWorld from '../HelloWorld.vue'

describe('HelloWorld.vue', () => {
  const msg = 'Hello World'
  it('renders correctly', () => {
    const wrapper = mount(HelloWorld, {
      props: {
        msg,
      },
    })
    expect(wrapper.text()).toContain(msg)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('counter increment exactly', async () => {
    const wrapper = mount(HelloWorld, {
      props: {
        msg,
      },
    })
    expect(wrapper.find('button').text()).toContain('0')
    await wrapper.find('button').trigger('click')
    expect(wrapper.find('button').text()).toContain('1')
  })
})
