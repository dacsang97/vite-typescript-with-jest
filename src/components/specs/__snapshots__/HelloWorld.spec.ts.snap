// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders correctly 1`] = `"<h1>Hello World</h1><button>count is: 0</button><p> Edit <code>components/HelloWorld.vue</code> to test hot module replacement. </p>"`;
